ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,82 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            IHqlExpression * c2 = queryRealChild(expr, 2);
+            IValue * t2 = c2 ? c2->queryValue() : NULL;
+            if (t0 && (!c2 || t2))
+            {
+                if(isUnicodeType(t0->queryType()))
+                {
+                    HqlExprArray results;
+                    unsigned plen = t0->queryType()->getStringLen();
+                    UChar * pattern = (UChar *)malloc((plen+1)*2);
+                    t0->getUCharStringValue(plen+1, pattern); //plen+1 so get null-terminated
+                    ICompiledUStrRegExpr * compiled = rtlCreateCompiledUStrRegExpr(pattern, !expr->hasAttribute(noCaseAtom));
+                    if (t1 && compiled)
+                    {
+                        unsigned slen = t1->queryType()->getStringLen();
+                        UChar * search = (UChar *)malloc((slen)*2);
+                        t1->getUCharStringValue(slen, search);
+
+                        IUStrRegExprFindInstance * match = compiled->find(search, 0, slen);
+                        while (match->found())
+                        {
+                            Owned<ITypeInfo> type;
+                            size32_t len;
+                            UChar * data;
+                            match->getMatchX(len, data, (unsigned)0);
+
+                            type.setown(makeUnicodeType(len,t0->queryType()->queryLocale()));
```

Comment:
style: More typical to say 

Owned<ITypeInfo> type = makeUnicodeType(len,t0->queryType()->queryLocale());